org.adempiere.base/src/org/adempiere/model/GenericPO.java

```diff
@@ -132,6 +146,604 @@ protected int get_AccessLevel() {
+	/** Column name C_DocType_ID */
+	public static final String COLUMNNAME_C_DocType_ID = "C_DocType_ID";
+	/** Column name DateAcct */
+	public static final String COLUMNNAME_DateAcct = "DateAcct";
+	/** Column name DateTrx */
+	public static final String COLUMNNAME_DateTrx = "DateTrx";
+	/** Column name IsApproved */
+	public static final String COLUMNNAME_IsApproved = "IsApproved";
+	/** Column name Processed */
+	public static final String COLUMNNAME_Processed = "Processed";
+	/** Column name DocumentNo */
+	public static final String COLUMNNAME_DocumentNo = "DocumentNo";
+	/** Column name Description */
+	public static final String COLUMNNAME_Description = "Description";
+	/** Column name SalesRep_ID */
+	public static final String COLUMNNAME_SalesRep_ID = "SalesRep_ID";
+	/** Column name AD_User_ID */
+	public static final String COLUMNNAME_AD_User_ID = "AD_User_ID";
+	/** Column name C_Currency_ID */
+	public static final String COLUMNNAME_C_Currency_ID = "C_Currency_ID";
+
+	/** DocStatus AD_Reference_ID=131 */
+	public static final int DOCSTATUS_AD_Reference_ID=131;
+	/** Drafted = DR */
+	public static final String DOCSTATUS_Drafted = "DR";
+	/** Completed = CO */
+	public static final String DOCSTATUS_Completed = "CO";
+	/** Approved = AP */
+	public static final String DOCSTATUS_Approved = "AP";
+	/** Not Approved = NA */
+	public static final String DOCSTATUS_NotApproved = "NA";
+	/** Voided = VO */
+	public static final String DOCSTATUS_Voided = "VO";
+	/** Invalid = IN */
+	public static final String DOCSTATUS_Invalid = "IN";
+	/** Reversed = RE */
+	public static final String DOCSTATUS_Reversed = "RE";
+	/** Closed = CL */
+	public static final String DOCSTATUS_Closed = "CL";
+	/** Unknown = ?? */
+	public static final String DOCSTATUS_Unknown = "??";
+	/** In Progress = IP */
+	public static final String DOCSTATUS_InProgress = "IP";
+	/** Waiting Payment = WP */
+	public static final String DOCSTATUS_WaitingPayment = "WP";
+	/** Waiting Confirmation = WC */
+	public static final String DOCSTATUS_WaitingConfirmation = "WC";
+	/**
+	 * Set Document Status.
+	 *	@param DocStatus The current status of the document
+	 */
+	public void setDocStatus(String DocStatus) {
+		columnExists(COLUMNNAME_DocStatus, true);
+		set_Value (COLUMNNAME_DocStatus, DocStatus);
+	}
+
+	/**
+	 * Get Document Status.
+	 * @return The current status of the document
+	 */
+	public String getDocStatus() {
+		columnExists(COLUMNNAME_DocStatus, true);
+		return (String)get_Value(COLUMNNAME_DocStatus);
+	}
+
+	/**************************************************************************
+	 * 	Process document
+	 *	@param processAction document action
+	 *	@return true if performed
+	 */
+	public boolean processIt (String processAction) {
+		m_processMsg = null;
+		DocumentEngine engine = new DocumentEngine (this, getDocStatus());
+		return engine.processIt (processAction, getDocAction());
+	}	//	process

```

Comment:
I think it probably make sense to have GenericPO implement the DocAction interface and has add "@Override" annotation to this methods.

Reply:
added the @Override declaration to all methods